src/parquet/column/levels.h

```diff
@@ -123,8 +132,8 @@ class LevelDecoder {
   // Decodes a batch of levels into an array and returns the number of levels decoded
   size_t Decode(size_t batch_size, int16_t* levels) {
     size_t num_decoded = 0;
-    if (!rle_decoder_ && !bit_packed_decoder_) {
-      throw ParquetException("Level decoders are not initialized.");
+    if (num_values_remaining_ == 0) {
+      throw ParquetException("Level decoders are not initialized with data.");
     }
```

Comment:
That seems fine. Can you add a test case with an empty data page (num_values == 0) to make sure this returns 0?